packages/start/src/server-handler/index.tsx

```diff
@@ -23,28 +26,44 @@ async function handleServerAction(event: H3Event) {
 export async function handleServerRequest(request: Request, _event?: H3Event) {
   const method = request.method
   const url = new URL(request.url, 'http://localhost:3000')
+  // extract the serverFnId from the url as host/_server/:serverFnId
+  // Define a regex to match the path and extract the :thing part
+  const regex = /\/_server\/([^/?#]+)/
+
+  // Execute the regex
+  const match = url.pathname.match(regex)
+  const serverFnId = match ? match[1] : null
   const search = Object.fromEntries(url.searchParams.entries()) as {
-    _serverFnId?: string
-    _serverFnName?: string
     payload?: any
   }
 
-  const serverFnId = search._serverFnId
-  const serverFnName = search._serverFnName
+  invariant(typeof serverFnId === 'string', 'Invalid server action')
 
-  if (!serverFnId || !serverFnName) {
-    throw new Error('Invalid request')
-  }
+  const serverFnInfo = serverFnManifest[serverFnId]
 
-  invariant(typeof serverFnId === 'string', 'Invalid server action')
+  invariant(serverFnInfo, 'Server function not found')
 
   if (process.env.NODE_ENV === 'development')
-    console.info(`ServerFn Request: ${serverFnId} - ${serverFnName}`)
-  if (process.env.NODE_ENV === 'development') console.info()
+    console.info(`\nServerFn Request: ${serverFnId}`)
+
+  let action: Function
+  if (process.env.NODE_ENV === 'development') {
+    action = (await (globalThis as any).app
+      .getRouter('server')
```

Comment:
Is this `.getRouter` call a Vinxi thing? or is it from Nitro?

Best to leave a comment here with any reference links that are available.